src/main/java/mike/Storage.java

```diff
@@ -0,0 +1,74 @@
+package mike;
+
+import mike.task.Task;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Used to store or load from the task lists into the disk
+ */
+public class Storage {
+    Path filePath;
+
+    /**
+     * Creates a Storage object corresponding to the path
+     *
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = Paths.get(filePath);
+    }
+
+    /**
+     * Saves the taskList into the file on the disk corresponding to the filePath
+     *
+     * @param taskList the task list based on the user's input
+     * @throws IOException
+     */
+    public void save(TaskList taskList) throws IOException {
+        Path dirPath = filePath.getParent();
```

Comment:
I like how the variable names are not misleading.

Reply:
I like that it throws the standard IOException for errors, it helps that it's not misleading, as @yeoutzer said!